### DISCUSSION TITLE
Can I add a static image to the background of an animation?

### DISCUSSION THREAD

**Discussion by arrowtype** [coldtype/coldtype#129] [Q&A] [answered]
I’m attempting to place a background image (i.e. for a border), then have an animation run over top of it.

![handwriting_latin](https://user-images.githubusercontent.com/45946693/217682523-58cd2dd3-6896-4e8f-af45-3a7a2a0232f9.gif)

[Here’s my current attempt...](https://github.com/arrowtype/shantell-sans/blob/eed8b17a3677a2ceee7c510b3954a66526fdd93f/sources/04-coldtype/basic-loop-w_background_image.coldtype.py)

I’m not quite sure if adding a background image is possible, but I assume it probably is. I would be happy to insert it as either a PNG or (as a stretch goal) vectors/SVG code.

If so, what might be the best way to do so... drawbot? An coldtype img() function? If these are possible... where should I place them, and how? I’ve experimented a bit, but haven’t yet had success.

I don’t quite understand that paradigms here, I guess, so it’s hard to intelligently guess at where to add in drawbot functions, or how I can use the information in the docs at https://coldtype.goodhertz.com/classes_functions.html

In general, I’d love to explore how I could get further into “graphic design” / motion design layouts with ColdType, to produce more than animated text alone. Thanks so much for any advice here!

**Comment by stenson** [comment, marked as answer]
The image support in Coldtype isn't one of its strengths, though there is some for sure. Undocumented (as usual), but the `SkiaImage` class is probably the best bet, just because it provides high-quality resizing, which you'd need to get from the 16x9 image to the 1920x1080 background. Here's a slightly modified version of your code w/ a SkiaImage: https://github.com/stenson/shantell-sans/blob/main/sources/04-coldtype/basic-loop-w_background_image.coldtype.py

As for drawBot, while it is possible to use drawBot and Coldtype together, it's not exactly a two-way street, since the rendering paradigms are different. In drawBot you can "render" at any point in the code, but in Coldtype the only way to get something to show up is to return it from the function. So while you could render a Coldtype `P` vector using drawBot as the rendering engine (as in this example: https://github.com/coldtype/coldtype/blob/main/examples/drawbot/both.py), you can't conversely use an `img` call in a Coldtype-style rendering. To do that, you'll need to do something like what's happening with the .insert calls at the bottom of the modified basic-loop source linked above.

All that said, I think for something like this — placing an image, modifying color or contrast, adding music, etc. — I almost always end up doing that in a video editor (like Davinci Resolve) or something like Blender, since I tend to think of Coldtype as an instrument rather than as a DAW. Doesn't mean it's the only approach, but probably explains why some more standard graphic design-y features are missing / obscure in Coldtype itself.

Maybe not a totally satisfying answer to your question, but hopefully that helps somewhat!

**Reply by arrowtype** [reply]
Ah, right! I had forgotten the approach of layering it into a video editor – I could see that making a lot of sense here. 

Thanks so much also for the helpful example! That is indeed working really well.

Thank you also for the helpful explanation of the Coldtype vs DrawBot paradigms – that really helps me parse your example.

If you don’t mind me asking, I’m not finding answers on google ... where might I learn more about how you are using `__sibling__`, as at [line 21](https://github.com/stenson/shantell-sans/blob/b27eb7e70d20e81288216b9656b722fce286d214/sources/04-coldtype/basic-loop-w_background_image.coldtype.py#L21) of your example?

```
img_src = __sibling__("../../specimens/shantell_sans-hero_16x9.png")
```

**Reply by stenson** [reply]
Happy to help! Ah right, `__sibling__`... maybe not my best idea — it's a generic/"magic" function I added to coldtype for a couple of reasons.

Given the way coldtype loads source files (via `run_path` on a tempfile), the `__file__` variable isn't available/correct. I do provide a `__FILE__` substitute for when you'd like to access `__file__` (does the same thing, just a different case), but then the idiom of `Path(__FILE__).parent / "my_relative_file.png")` looks pretty weird, so I thought, why not automatically populate a function called `__sibling__` that lets you do `__sibling__("my_relative_file.png")` as an equivalent to the longer Path(__FILE__) one?

All that said, actually kind of unnecessary to use `__sibling__` in the instance above, since you have to backtrack with `../..` anyway. So I think that could actually just be `img_src = Path("specimens/shantell_sans-hero_16x9.png")` if you're always running the coldtype command from the root of the project. (If you aren't, then something like __sibling__ is helpful.)
